.gitignore

```diff
@@ -1,7 +1,22 @@
 *.o
 *.a
 *~
+*.so
 led-image-viewer
 demo
 minimal-example
 text-example
+
+# visual studio
```

Comment:
All these files don't look like they would be generated if the Makefile in this project is being used, so we don't need to mention them here. Probably the only file we need to mention is *.dll